src/main/kotlin/edu/ics499/team1/app/services/TripService.kt

```diff
@@ -1,11 +1,40 @@
 package edu.ics499.team1.app.services
 
 import edu.ics499.team1.app.domains.Trip
+import edu.ics499.team1.app.domains.VehicleTripMileage
+import edu.ics499.team1.app.entities.TripEntity
 import edu.ics499.team1.app.repositories.TripRepository
+import edu.ics499.team1.app.repositories.VehicleRepository
 
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
 
 @Service
-class TripService(private val dataSource: TripRepository) {
-    fun getTrips(): Collection<Trip> = dataSource.findAll()
+class TripService(
+    private val tripRepository: TripRepository,
+    private val vehicleRepository: VehicleRepository
+) {
+
+    fun getTrips(): List<TripEntity> = tripRepository.findAll()
+
+    fun addTrip(trip: Trip): TripEntity {
+        val vehicle = vehicleRepository.getReferenceById(trip.vehicleId)
+
+        return tripRepository.save(trip.toTripEntity(vehicle))
+    }
+
+    @Transactional
+    fun setMileage(tripId: Int, mileage: Int) {
+        tripRepository.updateMileage(tripId, mileage)
+    }
```

Comment:
I cannot remember why, but this failed so I added Transactional. I know the error said something about transactions. Searching on google provided this annotation and it works 🤷